src/components/Canvas/Editor.tsx

```diff
@@ -569,6 +586,56 @@ export default class Editor extends EventDispatcher {
     this.interactionLayer.setDpr(dpr);
   }
 
+  /**
+   * @summary Sets the data of the pixel array (use with caution, since data will be overwritten!)
+   * @param data Array of PixelModifyItem (It must be a rectangular array, i.e. all rows must have the same length)
+   */
+  setData(data: Array<Array<PixelModifyItem>>) {
+    const { isDataValid, rowCount, columnCount } = validateSquareArray(data);
+    if (!isDataValid) {
+      throw new Error(`Data is not valid`);
+    }
+    const leftColumnIndex = data[0][0].columnIndex;
+    const topRowIndex = data[0][0].rowIndex;
+
+    // reset data
+    this.dataLayer.setData(new Map());
+    for (let i = 0; i < data.length; i++) {
+      this.dataLayer.getData().set(topRowIndex + i, new Map());
+      for (let j = 0; j < data[i].length; j++) {
+        this.dataLayer
+          .getData()
+          .get(topRowIndex + i)!
+          .set(leftColumnIndex + j, { color: data[i][j].color });
+      }
+    }
+
+    this.gridLayer.setCriterionDataForRendering(this.dataLayer.getData());
+    this.gridLayer.setRowCount(rowCount);
+    this.gridLayer.setColumnCount(columnCount);
+
+    this.interactionLayer.setDataLayerRowCount(rowCount);
+    this.interactionLayer.setDataLayerColumnCount(columnCount);
+    this.emitDataChangeEvent({ data: this.dataLayer.getCopiedData() });
+    this.emitGridChangeEvent({
+      dimensions: this.dataLayer.getDimensions(),
+      indices: this.dataLayer.getGridIndices(),
+    });
+    this.interactionLayer.setCriterionDataForRendering(
+      this.dataLayer.getData(),
+    );
+    this.dataLayer.setCriterionDataForRendering(this.dataLayer.getData());
+    this.interactionLayer.resetCapturedData();
+
+    this.setPanZoom({
+      offset: {
+        x: -this.panZoom.scale * (columnCount / 2) * this.gridSquareLength,
+        y: -this.panZoom.scale * (rowCount / 2) * this.gridSquareLength,
+      },
+    });
```

Comment:
extending the question for intention of this hook, why do we reset panzoom to middle when colorPixels doesn't

Reply:
I intended to make the pixel canvas centered when set Data is called. However, this is just my personal opinion🥸